### ISSUE TITLE
Invisible message ID not functioning

### ISSUE THREAD

**Issue by OGAwal** [lebaston100/MIDItoOBS#1]
Hello. First off, thank you for this script. It is what I was looking for. I setup a midi button to "hide" a render source in OBS, but the script is not functioning. It registers the press, but fails to hide the source in OBS. Also, is there a modification I can implement that will give me toggle abilities? For instance, assigning "visible" and "invisible" to the same midi button and have it toggle between them? Thank you for your hard work.

**Comment by lebaston100** [comment]
Thanks for your feedback. I will investigate this. 

For the future there are some changes in the recent websocket plugin release that i have to implement but this will be a few weeks.

**Comment by lebaston100** [comment]
I pushed a fix for this. Can you download it again and try it? Waiting for your response.

**Comment by OGAwal** [comment]
Setup.py is crashing. See screen cap. 

[https://gyazo.com/70ed0928f3ff37c3fb81b8391f11c7f7](url)

**Comment by OGAwal** [comment]
Sorry. See this link... [https://imgur.com/a/GwINF](url)

**Comment by lebaston100** [comment]
Well, that's a new one. shouldn't have anything to do with the other bugfix. Is the device still open in another application?

**Comment by OGAwal** [comment]
That was the cause. I have the controller attached to voicemeeter. I closed it down setup.py is working. It seems like the hotfix is working. It is hiding/unhiding the sources appropriately. Thank you. Do you know if there is a way to modify the config file to attach 2 sources to the same midi key?

**Comment by lebaston100** [comment]
Now there is one :D Download your custom build from here: https://drive.google.com/open?id=1j6mR7vZGPm-ErW7JxjzYAq3GxUlhBki_
How to setup: 
Set the functions up as normal on 2 different buttons. Now open the config with a text editor.
Change the "msgNoC" value of the 2. button to the value of the first button: Here is a example:
Before: https://imgur.com/a/m2aje
After: https://imgur.com/a/XFXYo
Now start the main.py as normal and it should send it at the same time.
There is no limit on how many different functions you can assign. You should even be able to mix different ones but this is not tested.

**Comment by OGAwal** [comment]
Wow thank you. Your link is giving me a 404 error. Also, just as a question of curiosity... The script you wrote has the ability to toggle mute in OBS. This ability is not available for rendering sources... is that an OBS limitation? I don't know coding, so please pardon my ignorance.

**Comment by lebaston100** [comment]
Try again as i had edited the link earlier. 
I basically just use the api provided by the websocket plugin as seen here: https://github.com/Palakis/obs-websocket/blob/master/docs/generated/protocol.md
It has a function to toggle the mute status of a source but not for visibility. As i said earlier there is some work to do and i will add this to the todo list.

**Comment by lebaston100** [comment]
I will close this issue because the problem is fixed, Feel free to contact me again if there are any problems.

**Comment by lebaston100** [comment]
Just to give you an update with the latest update i've implemented the macro support and a toogle for the visibility.

**Comment by OGAwal** [comment]
Hi lebaston! I just wanted to let you know I got your message about the update. I appreciate your on going effort with this plugin. It still is the only thing I can find that functions well. With the update to "togglesourcevisibility"... it works, but it doesnt. I tried to set 2 sources to toggle on the same button. I want one to become visible and the other to become invisible with the same key press. Though I change the script appropriately, it doesn't work. Any easy fix to that?

**Comment by lebaston100** [comment]
I will try it and get back to you.

**Comment by lebaston100** [comment]
Found the bug and hopefully fixed it now with the last commit.

**Comment by OGAwal** [comment]
Works amazing! Thank you very much. I would love to post this on the OBS forums with your permission. Should it work with most midi devices?

**Comment by lebaston100** [comment]
Thanks for letting me know. I can't fix what I don't know.
I have thought about posting it in the scripts section of the forum but i won't do that.
But feel free to suggest it if someone asks in the forum or discord or wherever.
